mailbox/maildir/src/test/java/org/apache/james/mailbox/maildir/MaildirMailboxManagerProvider.java

```diff
@@ -21,41 +21,36 @@
 
 import java.io.IOException;
 
-import org.apache.james.mailbox.acl.GroupMembershipResolver;
 import org.apache.james.mailbox.acl.MailboxACLResolver;
-import org.apache.james.mailbox.acl.SimpleGroupMembershipResolver;
 import org.apache.james.mailbox.acl.UnionMailboxACLResolver;
 import org.apache.james.mailbox.exception.MailboxException;
-import org.apache.james.mailbox.store.Authenticator;
-import org.apache.james.mailbox.store.Authorizator;
 import org.apache.james.mailbox.store.JVMMailboxPathLocker;
+import org.apache.james.mailbox.store.MailboxManagerOptions;
 import org.apache.james.mailbox.store.StoreMailboxAnnotationManager;
 import org.apache.james.mailbox.store.StoreMailboxManager;
 import org.apache.james.mailbox.store.StoreRightManager;
 import org.apache.james.mailbox.store.event.DefaultDelegatingMailboxListener;
 import org.apache.james.mailbox.store.event.MailboxEventDispatcher;
 import org.apache.james.mailbox.store.mail.model.DefaultMessageId;
-import org.apache.james.mailbox.store.mail.model.impl.MessageParser;
 import org.junit.rules.TemporaryFolder;
 
 public class MaildirMailboxManagerProvider {
-    public static StoreMailboxManager createMailboxManager(String configuration, TemporaryFolder temporaryFolder) throws MailboxException, IOException {
+    public static StoreMailboxManager createMailboxManager(String configuration, TemporaryFolder temporaryFolder,
+                                                           MailboxManagerOptions options) throws MailboxException, IOException {
```

Comment:
what's the point of this option stuff if it's still adding some parameters to methods, instead of removing?

Reply:
The goal is to say: now I can instanciate  whatever maildir manager with whatever componant configued, without :
 - introducing several provider methods
 - changing 1000  constructors

Note that this change allow me to use the provider in mpt tests and avoid a lot of burden.

The idea of this commit is to remove coupling. Coupling between every test that uses a manager and the constructor itself. Now when I change the constructor of the manager, I need to change a way more limited number of class (test class not using the given added parameter will not need to change, at all!)